navigation-material/src/main/java/com/google/accompanist/navigation/material/BottomSheetNavigator.kt

```diff
@@ -165,63 +177,78 @@ public class BottomSheetNavigator(
     /**
      * Access properties of the [ModalBottomSheetLayout]'s [ModalBottomSheetState]
      */
-    public val navigatorSheetState = BottomSheetNavigatorSheetState(sheetState)
+    val navigatorSheetState = BottomSheetNavigatorSheetState(sheetState)
 
     /**
      * A [Composable] function that hosts the current sheet content. This should be set as
      * sheetContent of your [ModalBottomSheetLayout].
      */
-    public val sheetContent: @Composable ColumnScope.() -> Unit = @Composable {
-        val columnScope = this
-        val saveableStateHolder = rememberSaveableStateHolder()
-        val backStackEntries by backStack.collectAsState()
-        val transitionsInProgressEntries by transitionsInProgress.collectAsState()
-
-        // We always replace the sheet's content instead of overlaying and nesting floating
-        // window destinations. That means that only *one* concurrent destination is supported by
-        // this navigator.
-        val latestEntry = backStackEntries.lastOrNull { entry ->
-            // We might have entries in the back stack that aren't started currently, so filter
-            // these
-            entry.lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED)
-        }
+    val sheetContent: @Composable ColumnScope.() -> Unit =
+        movableContentWithReceiverOf<ColumnScope> {
+            val saveableStateHolder = rememberSaveableStateHolder()
+            val transitionsInProgressEntries by transitionsInProgress.collectAsState()
 
-        // Mark all of the entries' transitions as complete, except for the entry we are
-        // currently displaying because it will have its transition completed when the sheet's
-        // animation has completed
-        DisposableEffect(backStackEntries) {
-            transitionsInProgressEntries.forEach {
-                if (it != latestEntry) state.markTransitionComplete(it)
+            val retainedEntry by produceState<NavBackStackEntry?>(initialValue = null) {
+                backStack
+                    .map { backStackEntries ->
+                        backStackEntries.lastOrNull { entry ->
+                            // We might have entries in the back stack that aren't started
+                            // currently, so filter these
+                            entry.lifecycle.currentState.isAtLeast(Lifecycle.State.INITIALIZED)
```

Comment:
@jbw0033 in tests (specifically, `testTapOnScrimDismissesSheetAndPopsBackStack`) the entry added would only be `INITIALIZED`. All is fine in the sample. Any ideas what I'm missing? Do we even need to filter?